[BUG] ssh tunnel seems killing itself because of concurrency

### Title

ssh tunnel seems killing itself because of concurrency

### Platform

Web - Edge

### Server Installation Method

Docker

### Version

1.11.1

### Troubleshooting

- [x] I have examined logs and tried to find the issue
- [x] I have reviewed opened and closed issues
- [x] I have tried restarting the application
- [x] I have read Known Issues (https://github.com/Termix-SSH/Support/issues/296)

### The Problem

Logs in docker:

[7:13:51 AM] [INFO] [📡] Tunnel creation request received [op:tunnel_create_request,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]
[7:13:51 AM] [INFO] [📡] Killing remote tunnel process [op:tunnel_remote_kill,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]
[7:13:51 AM] [INFO] [📡] Creating new SSH connection for tunnel [op:tunnel_connection_create,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]
[7:13:52 AM] [SUCCESS] [📡] **Tunnel port binding successful** [op:tunnel_port_bound,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]
[7:13:52 AM] [INFO] [📡] **Remote tunnel process found, proceeding to kill** [op:tunnel_remote_found,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]
[7:13:52 AM] [ERROR] [📡] SSH stderr for '2::0::host::43993::liangzhu::43991': write: Broken pipe
[7:13:56 AM] [SUCCESS] [📡] Remote tunnel process killed [op:tunnel_remote_killed,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]
[7:14:03 AM] [INFO] [📡] Killing remote tunnel process [op:tunnel_remote_kill,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]
[7:14:04 AM] [WARN] [📡] Remote tunnel process not found [op:tunnel_remote_not_found,user:xxx,host:2,tunnel:2::0::host::43993::liangzhu::43991]

Status in web browser:

<img width="855" height="276" alt="Image" src="https://github.com/user-attachments/assets/95f1ccfa-4479-41ab-80ea-bb0bcf8e2cc2" />

Maybe adding `await cleanupTunnelResources(tunnelName, true)` in connectSSHTunnel() would help.

### How to Reproduce

1. create a normal tunnel.
2. in the source server, manually using ssh -R or -L would work as espected. Set ~/.ssh/config if ProxyCommand is needed to connect to endpoint server.
3. using web client to setup tunnel would failed, showing logs above.


### Additional Context

By the way, using SOCKS5 Proxy to connect ssh server works well when connect ssh via browser. However servers would keep at "offline" status (the red dot). Also Android termix apps in this case seems cannot connect to those ssh server using proxy. So in the context I use tun-proxy container to provide network for termix and everything works fine.

```yaml
services:
  # 1. The Transparent Proxy Sidecar
  tun-proxy:
    image: xjasonlyu/tun2socks:v2.6.0
    container_name: termix-proxy
    restart: unless-stopped
    cap_add:
      - NET_ADMIN
    cap_drop:
      - ALL
    devices:
      - '/dev/net/tun:/dev/net/tun'
    # If SOCKS proxy requires auth, use: socks5://user:pass@host.docker.internal:PORT
    environment:
      - PROXY=socks5://host.docker.internal:<SOCKS5-proxy-port>
      - LOGLEVEL=info
      - MTU=1380
    extra_hosts:
      - "host.docker.internal:host-gateway"
    ports:
      # Port bindings MUST move to the sidecar container
      - '127.0.0.1:<port-for-reverse-proxy>:8080'
    networks:
      termix_proxy_net: # Connect to our named bridge

  # 2. Application Container
  termix:
    image: bugattiguy527/termix:latest
    container_name: termix
    restart: unless-stopped
    # Attach this container directly to the proxy's network namespace
    network_mode: "service:tun-proxy"
    depends_on:
      - tun-proxy
    volumes:
      - termix-data:/app/data
    # Environment variables are kept for any internal tools that might still want them,
    # but all raw TCP/UDP is now captured by the sidecar.
    environment:
      - PORT=8080
      - HTTP_PROXY=
      - http_proxy=
      - HTTPS_PROXY=
      - https_proxy=

volumes:
  termix-data:
    driver: local
```

--- a/README.md
+++ b/README.md
@@ -169,6 +169,10 @@ networks:
   <a href="https://akamai.com/">
     <img src="https://upload.wikimedia.org/wikipedia/commons/8/8b/Akamai_logo.svg" height="50" alt="Akamai">
   </a>
+  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+  <a href="https://aws.amazon.com/">
+    <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/93/Amazon_Web_Services_Logo.svg/960px-Amazon_Web_Services_Logo.svg.png" height="50" alt="AWS">
+  </a>
 </p>
 
 # Support

--- a/src/backend/ssh/tunnel.ts
+++ b/src/backend/ssh/tunnel.ts
@@ -546,7 +546,7 @@ async function connectSSHTunnel(
 
   tunnelConnecting.add(tunnelName);
 
-  cleanupTunnelResources(tunnelName, true);
+  await cleanupTunnelResources(tunnelName, true);
 
   if (retryAttempt === 0) {
     retryExhaustedTunnels.delete(tunnelName);

